flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mongodb-cdc/src/test/java/com/ververica/cdc/connectors/mongodb/source/MongoDBFullChangelogITCase.java

```diff
@@ -188,6 +205,261 @@ public void testJobManagerFailoverSingleParallelism() throws Exception {
+                        "+I[15213, user_15213, Shanghai, 123567891234]",
+                        "-U[2000, user_21, Shanghai, 123567891234]",
+                        "+U[2000, user_21, Pittsburgh, 123567891234]",
+                        "-D[1019, user_20, Shanghai, 123567891234]");
+        // when skip backfill, the wal log between (snapshot, high_watermark) will be seen as
+        // stream event.
+        assertEqualsInAnyOrder(expectedRecords, records);
+    }
+
+    @Test
+    public void testSkipBackfillWithDMLPostLowWaterMark() throws Exception {
+        if (!parallelismSnapshot) {
+            return;
+        }
+
+        List<String> records =
+                getResultWithDMLInSnapshotPhase(true, 25, USE_POST_LOWWATERMARK_HOOK, true);
+
+        List<String> expectedRecords =
+                Arrays.asList(
+                        "+I[101, user_1, Shanghai, 123567891234]",
+                        "+I[102, user_2, Shanghai, 123567891234]",
+                        "+I[103, user_3, Shanghai, 123567891234]",
+                        "+I[109, user_4, Shanghai, 123567891234]",
+                        "+I[110, user_5, Shanghai, 123567891234]",
+                        "+I[111, user_6, Shanghai, 123567891234]",
+                        "+I[118, user_7, Shanghai, 123567891234]",
+                        "+I[121, user_8, Shanghai, 123567891234]",
+                        "+I[123, user_9, Shanghai, 123567891234]",
+                        "+I[1009, user_10, Shanghai, 123567891234]",
+                        "+I[1010, user_11, Shanghai, 123567891234]",
+                        "+I[1011, user_12, Shanghai, 123567891234]",
+                        "+I[1012, user_13, Shanghai, 123567891234]",
+                        "+I[1013, user_14, Shanghai, 123567891234]",
+                        "+I[1014, user_15, Shanghai, 123567891234]",
+                        "+I[1015, user_16, Shanghai, 123567891234]",
+                        "+I[1016, user_17, Shanghai, 123567891234]",
+                        "+I[1017, user_18, Shanghai, 123567891234]",
+                        "+I[1018, user_19, Shanghai, 123567891234]",
+                        "+I[2000, user_21, Pittsburgh, 123567891234]",
+                        "+I[15213, user_15213, Shanghai, 123567891234]",
+                        "+I[15213, user_15213, Shanghai, 123567891234]",
+                        "-U[2000, user_21, Shanghai, 123567891234]",
+                        "+U[2000, user_21, Pittsburgh, 123567891234]",
+                        "-D[1019, user_20, Shanghai, 123567891234]");
+        // when skip backfill, the wal log between (snapshot,  high_watermark) will still be
+        // seen as stream event. This will occur data duplicate. For example, user_20 will be
+        // deleted twice, and user_15213 will be inserted twice.
+        assertEqualsInAnyOrder(expectedRecords, records);
+    }
+
+    private List<String> getResultWithDMLInSnapshotPhase(
```

Comment:
withDML?  You can rename the method to `testSkipBackfillWhenWritingEvents( boolean skipBackFill, int fetchSize, int hookType, boolean enableFullDocPrePostImage,  List<String> expectedRecords)`, and then you can gently  call this method in other tests